src/XIOPAK.CPP

```diff
@@ -1155,9 +1152,22 @@ doneOpen0:	/* come here on error opening file 1 */
 	return rv;
 
 }		/* xffilcmp */
-
-
-
+//=============================================================================
+SEC xfclear(	// Cleans the file by discarting all the data in the file
+	XFILE* xf)	// File to be clean
+	/* Returns: SECOK on sucess or
+				SECOTHER on failure
+	*/
+{
+	xlsterr(xf);				// clear any prior xiopak error (else addl calls will nop)
+	if ( (freopen(xf->xfname, "w", xf->han)) != nullptr			// Reopen with "w" to discard all the data in the file
+		&& ((freopen(xf->xfname, xf->access, xf->han)) != nullptr))	// Reopen with the initial mode
```

Comment:
What happens if the first reopen fails? Do you really want to try another reopen operation right away on the same file?

Reply:
If the first reopen fails then it should exit the function with an error.

Reply:
Will it exit before or after it tries the second reopen?

Reply:
I see. You're right.